iroh-net/src/magicsock.rs

```diff
@@ -1319,18 +1307,11 @@ impl MagicSock {
     ///
     /// Note this is a user-facing API and does not wrap the [`SocketAddr`] in a
     /// `QuicMappedAddr` as we do internally.
-    pub async fn get_mapping_addr(&self, node_key: &PublicKey) -> Option<SocketAddr> {
-        let (s, r) = tokio::sync::oneshot::channel();
-        if self
-            .inner
-            .actor_sender
-            .send(ActorMessage::GetMappingAddr(*node_key, s))
-            .await
-            .is_ok()
-        {
-            return r.await.ok().flatten().map(|m| m.0);
-        }
-        None
+    pub fn get_mapping_addr(&self, node_key: &PublicKey) -> Option<SocketAddr> {
```

Comment:
The new code gets the mapping addr immediately, the old code waits until all msgs currently in the queue are processed.

I guess this does not matter?

Reply:
Ah, that is a good question. Let's see what tests say. The relevant interaction though is `add_node_addr`, which also is sync. So I think it should be fine.